app/src/main/java/org/projectbuendia/client/ui/chart/PatientChartActivity.java

```diff
@@ -271,6 +281,28 @@ public void onPageFinished(WebView view, String url) {
                 mController.onAddTestResultsPressed();
             }
         });
+
+        initChartMenu();
+    }
+
+    private void initChartMenu() {
+        List<Chart> charts = mController.getCharts();
+        String[] menuArray = new String[charts.size()];
+        int i = 0;
+        for (Chart chart : charts) {
+            menuArray[i] = chart.name;
+            i++;
+        }
+        mAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, menuArray);
+        mDrawerList.setAdapter(mAdapter);
+        final Activity that = this;
+        mDrawerList.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                mController.updatePatientObsUi(position);
```

Comment:
The controller already handles the charts. I'm only building the list here because it's related to the view.
To change this is to start refactoring the this classes, something I was avoiding.